### PR TITLE
feat: PAYMENT_RECEIPT to have optional value.

### PR DESCRIPTION
**Pull request by andrehadianto**
[Notion link](notion-link-here) 

This PR adds... 
- remove mandatory `value` for `PAYMENT_RECEIPT`
- update `PolicyIdentifier` type

I think 
``` typescript
    if (textInputType !== "PAYMENT_RECEIPT" && !value) {
      throw new Error(ERROR_MESSAGE.MISSING_IDENTIFIER_INPUT);
    }
```
can be removed instead. Empty `value` will still be checked in the `regex` matching as well as BE validation

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [x] I've requested reviews from 1-2 reviewers
- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
- [ ] I've added/updated unit/integration tests
- [x] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
- [ ] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
- [ ] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)

### CODE REVIEW

File: src/utils/validateIdentifierInputs.ts [hunk 1]
```diff
@@ -62,6 +58,6 @@ export const validateIdentifierInputs = (
   ) {
     throw new Error(ERROR_MESSAGE.DUPLICATE_IDENTIFIER_INPUT);
   }
-
+  console.log("asdfasdf");
```
Comment (dalsontws) [inline src/utils/validateIdentifierInputs.ts:61]: need to remove the console.log here 😆

File: src/utils/validateIdentifierInputs.ts [hunk 2]
```diff
@@ -19,25 +19,21 @@ export const validateIdentifierInputs = (
...
-      textInputType === "PAYMENT_RECEIPT" &&
-      (!value ||
-        !isMatchRegex(
-          value,
-          validationRegex
-            ? validationRegex
-            : defaultPaymentReceiptValidationRegex
-        ))
-    ) {
-      throw new Error(ERROR_MESSAGE.INVALID_PAYMENT_RECEIPT_NUMBER);
```
Comment (dalsontws) [inline src/utils/validateIdentifierInputs.ts:32]: hmm actually this part is still required (can just remove the `!value` check), as invalid payment receipt will throw a specific error message, see `ERROR_MESSAGE.INVALID_PAYMENT_RECEIPT_NUMBER`

**Review by dalsontws — CHANGES_REQUESTED** [review]

**Review by andrehadianto — COMMENTED** [review]
press start review by mistake :|

File: src/utils/validateIdentifierInputs.ts (hunk 2 above)
Comment (dalsontws) [reply src/utils/validateIdentifierInputs.ts:32]: ah okay, understand! yup i think this will work, should we include a test case to check for empty payment receipt fields?

File: src/utils/validateIdentifierInputs.ts [hunk 3]
```diff
@@ -19,25 +19,21 @@ export const validateIdentifierInputs = (
...
-    }
-    if (!value) {
+    if (textInputType !== "PAYMENT_RECEIPT" && !value) {
```
Comment (dalsontws) [inline src/utils/validateIdentifierInputs.ts:22]: realised that altho the backend allows for optional fields for different text inputs types, this current logic only allows optional payment receipts. shall we address this in another story?

Comment (andrehadianto) [reply src/utils/validateIdentifierInputs.ts:22]: sounds good! But the PR of that story would be the same as the PR676?

Comment (ilhaoni) [reply src/utils/validateIdentifierInputs.ts:22]: or shall we address this inside PR676 haha

Comment (ilhaoni) [reply src/utils/validateIdentifierInputs.ts:22]: actually another question, is there a reason why we're doing an additional check here of  whether `textInputType !== "PAYMENT_RECEIPT"` haha

**Review by ilhaoni — COMMENTED** [review]
Left some comments!

File: src/utils/validateIdentifierInputs.ts (hunk 3 above)
Comment (andrehadianto) [reply src/utils/validateIdentifierInputs.ts:22]: as discussed,, need the check cause we only allow `PAYMENT_RECEIPT` to be empty currently!

Comment (ilhaoni) [reply src/utils/validateIdentifierInputs.ts:22]: ok!

**Review by ilhaoni — APPROVED** [review]
LGTM! :D

**Review by dalsontws — APPROVED** [review]
Thanks for working on this, LGTM!🚀
